src/test/fixed_point/rounding/rounding_fixed_point.cpp

```diff
@@ -24,5 +25,21 @@ namespace {
                 to_rep(rounding_fixed_point<>(-8)),
                 rounding_integer<>(-8)), "rounding_fixed_point ctor test failed");
         static_assert(rounding_fixed_point<>(0) == cnl::from_rep<rounding_fixed_point<>>{}(0), "rounding_fixed_point ctor test failed");
+        static_assert(identical(
+                to_rep(rounding_fixed_point<std::uint16_t, -8>(15.125)),
+                rounding_integer<std::uint16_t>{3872}), "rounding_fixed_point ctor test failed");
+
+        static_assert(identical(
+                to_rep(rounding_fixed_point<std::int32_t, -1>(0.249f)),
+                rounding_integer<std::int32_t>{0}), "rounding_fixed_point ctor test failed");
+        static_assert(identical(
+                to_rep(rounding_fixed_point<std::int32_t, -1>(0.25f)),
+                rounding_integer<std::int32_t>{1}), "rounding_fixed_point ctor test failed");
+    }
+
+    namespace test_conversion {
+        constexpr auto a = rounding_fixed_point<int, -4>{.4375};
```

Comment:
@hbe72 this is your test "constexprified". It was failing because `cnl::shift` wasn't specialized for `rounding_integer`. Hopefully that helps. If you want the free function equivalents, look in *cnl/rounding.h*.

Reply:
@johnmcfarlane Beautiful. Now I just need to figure out how this is done for rounding_elastic_number<Rep, Exponent> ;)

Reply:
Your tests pass currently. You may wish to review the branch again. Otherwise, I'll merged it soon.

Reply:
Ok. Can you check recently updated rounding_integer branch of the CDSP. I tried to implement rounding_elastic_number based on current develop of CNL. Nice list of errors I get ;)

Reply:
Try reversing `fixed_point` and `rounding_integer` so that `fixed_point` is on the outside. Alternatively, take a look at `static_number`. It is in early stages but contains all four types: `rounding_`, `overflow_`, `elastic_`, and `fixed_point`. You can turn off overflow checking with the `native_overflow_tag` (I think).